Source/WebKit/UIProcess/Model/ModelPresentationManagerProxy.mm

```diff
@@ -103,6 +104,17 @@
     m_activelyDraggedModelLayerIDs.clear();
 }
 
+void ModelPresentationManagerProxy::pageScaleDidChange(CGFloat newScale)
+{
+    for (auto& modelPresentation : m_modelPresentations.values()) {
+        if (RetainPtr modelView = modelPresentation->remoteModelView) {
+            CATransform3D newTransform = [modelView transform3D];
+            newTransform.m33 = newScale;
```

Comment:
we could add a comment explaining that this is safe because only the pageHostedView is part of the RemoteLayerTree (which can also set the transform)